### PR TITLE
add default /guard:cf

### PR DESCRIPTION
**Pull request by anporumb**

### CODE REVIEW

File: build_functions/CMakeLists.txt [hunk 1]
```diff
@@ -3,6 +3,9 @@
 
 set(build_c_tests_internal_dir ${CMAKE_CURRENT_LIST_DIR} CACHE INTERNAL "")
 
+#add_custom_target... below exists so that a project (called "c_build_functions") is created in Visual Studio so it can be edited easily in the IDE
```
Comment (dcristoloveanu) [inline build_functions/CMakeLists.txt:6]: >target... below exists s [](http://example.com/codeflow?start=12&length=24)

Love this!

**Review by dcristoloveanu — APPROVED** [review]
:shipit:

File: build_functions/CMakeLists.txt [hunk 2]
```diff
@@ -129,12 +132,13 @@ macro(set_default_build_options)
...
-        set(CMAKE_SHARED_LINKER_FLAGS "/INCREMENTAL:NO ${CMAKE_SHARED_LINKER_FLAGS} /LTCG /IGNORE:4075 /WX")
+        # /guard:cf is Enable Control Flow Guard https://docs.microsoft.com/en-us/cpp/build/reference/guard-enable-control-flow-guard, working for compiler and the linker
+        set(CMAKE_EXE_LINKER_FLAGS "/INCREMENTAL:NO ${CMAKE_EXE_LINKER_FLAGS} /LTCG /IGNORE:4075 /WX /guard:cf")
+        set(CMAKE_SHARED_LINKER_FLAGS "/INCREMENTAL:NO ${CMAKE_SHARED_LINKER_FLAGS} /LTCG /IGNORE:4075 /WX /guard:cf")
         set(CMAKE_STATIC_LINKER_FLAGS "${CMAKE_STATIC_LINKER_FLAGS} /WX")
 
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /GL")
-        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /GL")
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /GL /guard:cf")
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /GL /guard:cf")
```
Comment (dcristoloveanu) [inline build_functions/CMakeLists.txt:141]: >set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /GL /guard:cf") [](http://example.com/codeflow?start=8&length=51)

This makes me think that binskim should actually fail on the EBS repo.

Comment (dcristoloveanu) [reply build_functions/CMakeLists.txt:141]: Maybe we need a task to make sure binskim IS running on the correct binaries (in Elasting Log is was actually being ran on the source folder)

Comment (anporumb) [reply build_functions/CMakeLists.txt:141]: [https://msazure.visualstudio.com/One/_sprints/taskboard/Azure%20Messaging%20GeoReplication/One/Copper/CY22Q3/Weekly/Wk25](https://msazure.visualstudio.com/One/_sprints/taskboard/Azure%20Messaging%20GeoReplication/One/Copper/CY22Q3/Weekly/Wk25)

created

**Pull request merged**